# Trying out different sort methods

In [1]:
import time
import random

# Set seed.
random.seed(a=100)

# Create our default list.
short_list = list(random.sample(range(1000000), 10))
long_list = list(random.sample(range(1000000), 10000))

### Heapsort

In [2]:
from heapq import heappush, heappop
def heapsort(iterable):
    h = []
    for value in iterable:
        heappush(h, value)
    return [heappop(h) for i in range(len(h))]

In [5]:
start_time = time.time()

heapsort(short_list)

print("--- %s seconds ---" % (time.time() - start_time))

--- 0.0069789886474609375 seconds ---


In [6]:
start_time = time.time()

heapsort(long_list)

print("--- %s seconds ---" % (time.time() - start_time))

--- 0.024034976959228516 seconds ---


### Selection Sort

In [7]:
def selection_sort(a_list):
    for i in range(len(a_list)):
        # Find the minimum element in remaining
        minPosition = i
        
        for j in range(i+1, len(a_list)):
            if a_list[minPosition] > a_list[j]:
                minPosition = j
        # Swap the found minimum element with minPosition
        temp = a_list[i]
        a_list[i] = a_list[minPosition]
        a_list[minPosition] = temp
        
    return a_list

In [8]:
start_time = time.time()

selection_sort(long_list)

print("--- %s seconds ---" % (time.time() - start_time))

--- 8.215842008590698 seconds ---


In [9]:
start_time = time.time()

selection_sort(short_list)

print("--- %s seconds ---" % (time.time() - start_time))

--- 0.0001678466796875 seconds ---


### Conclusion

It looks like selection sort took a while sorting our long list with 8.2 seconds. Heapsort did fine with sorting long list. But overall, the sort method buili in python is still the fasted (and by far the easiest method to use). 